drf_spectacular/contrib/rest_framework_recursive.py

```diff
@@ -1,9 +1,18 @@
+from rest_framework import serializers
+
 from drf_spectacular.extensions import OpenApiSerializerFieldExtension
+from drf_spectacular.plumbing import build_array_type
 
 
 class RecursiveFieldExtension(OpenApiSerializerFieldExtension):
     target_class = "rest_framework_recursive.fields.RecursiveField"
 
     def map_serializer_field(self, auto_schema, direction):
-        component = auto_schema.resolve_serializer(self.target.proxied, direction)
+        proxied = self.target.proxied
+
+        if isinstance(proxied, serializers.ListSerializer):
```

Comment:
since you aready use plumbing, you could have used `is_list_serializer(proxied)` instead

Reply:
Sure. Pushed the change.